scripts/check-node-versions.mjs

```diff
@@ -24,46 +91,67 @@ if (!nodeVersion) {
 	);
 	process.exit(1);
 } else {
-	log(`Found node version ${nodeVersion} in \`.nvmrc\``);
+	log(`Found node version ${showSemverVersion(nodeVersion)} in \`.nvmrc\``);
 }
 
 const requiredNodeVersionMatches =
-	/** @type {const} @satisfies {ReadonlyArray<{filepath: string, pattern: RegExp}>}*/ ([
+	/** @type {const} @satisfies {ReadonlyArray<{filepath: string, pattern: RegExp, matchLevel: MatchLevel}>}*/ ([
 		{
 			filepath: 'dotcom-rendering/Containerfile',
 			pattern: /^FROM node:(.+)-alpine$/m,
+			matchLevel: 'Patch',
 		},
 		{
 			filepath: 'dotcom-rendering/scripts/deploy/riff-raff.yaml',
 			pattern: /^ +Recipe: dotcom-rendering.*-node-(\d+\.\d+\.\d+)$/m,
+			matchLevel: 'Patch',
 		},
 		{
 			filepath: 'apps-rendering/riff-raff.yaml',
 			pattern: /^ +Recipe: .+-mobile-node(\d+\.\d+\.\d+).*$/m,
+			matchLevel: 'Patch',
 		},
 		{
 			filepath: 'dotcom-rendering/package.json',
 			pattern: /^\s*"@types\/node": "(\d+\.\d+\.\d+)",?$/m,
+			matchLevel: 'Minor',
 		},
 		{
 			filepath: 'apps-rendering/package.json',
 			pattern: /^\s*"@types\/node": "(\d+\.\d+\.\d+)",?$/m,
+			matchLevel: 'Minor',
 		},
 	]);
 
 const problems = (
 	await Promise.all(
-		requiredNodeVersionMatches.map(async ({ filepath, pattern }) => {
+		requiredNodeVersionMatches.map(async ({ filepath, pattern, matchLevel }) => {
 			const fileContents = await readFile(
 				resolve(...filepath.split('/')),
 				'utf-8',
 			);
 			const foundNodeVersion =
 				fileContents.match(pattern)?.[1] ?? undefined;
+			
+			if (foundNodeVersion !== undefined) {
+				const parsedVersion = parseNodeVersion(foundNodeVersion);
+
+				if (parsedVersion !== undefined) {
+					const matchError = matchVersions(nodeVersion, parsedVersion, matchLevel);
+
+					if (matchError !== undefined) {
+						return [
+							`Node version in ${filepath} (${showSemverVersion(parsedVersion)}) `,
+							`does not match \`.nvmrc\` (${showSemverVersion(nodeVersion)}): `,
+							matchError
+						].join('');
+					}
+
+					return undefined;
+				}
+			}
```

Comment:
This may not be the best place to discuss code style, [enforcing this via a linting rule may be favourable](https://github.com/guardian/dotcom-rendering/pull/8784).